server/src/immich/app.service.ts

```diff
@@ -28,4 +59,47 @@ export class AppService {
     await this.serverService.handleVersionCheck();
     this.logger.log(`Feature Flags: ${JSON.stringify(await this.serverService.getFeatures(), null, 2)}`);
   }
+
+  ssr(excludePaths: string[]) {
+    const index = readFileSync('/usr/src/app/www/index.html').toString();
+
+    return async (req: Request, res: Response, next: NextFunction) => {
+      if (
+        req.url.startsWith('/api') ||
+        req.method.toLowerCase() !== 'get' ||
+        excludePaths.find((item) => req.url.startsWith(item))
+      ) {
+        return next();
+      }
+
+      const targets = [
+        {
+          regex: /^\/share\/(.+)$/,
+          onMatch: async (matches: RegExpMatchArray) => {
+            const key = matches[1];
+            const auth = await this.authService.validateSharedLink(key);
+            return this.sharedLinkService.getMetadataTags(auth);
+          },
+        },
+      ];
+
+      let html = index;
+
+      try {
+        for (const { regex, onMatch } of targets) {
+          const matches = req.url.match(regex);
+          if (matches) {
+            const meta = await onMatch(matches);
+            if (meta) {
+              html = render(index, meta);
+            }
+
+            break;
+          }
+        }
+      } catch {}
+
+      res.type('text/html').header('Cache-Control', 'no-store').send(html);
```

Comment:
I thought disabling the cache was only for testing?

Reply:
Oh no, I think it is actually better to just never cache the index page, since it will be dynamic sometimes.